python/pyspark/sql/functions.py

```diff
@@ -15524,9 +15524,10 @@ def udtf(
     ----------
     cls : class
         the Python user-defined table function handler class.
-    returnType : :class:`pyspark.sql.types.StructType` or str
+    returnType : :class:`pyspark.sql.types.StructType` or str, optional
         the return type of the user-defined table function. The value can be either a
         :class:`pyspark.sql.types.StructType` object or a DDL-formatted struct type string.
+        If None, the handler class must provide `analyze` static method.
```

Comment:
Can you please port the description of how the 'analyze' method works from the PR description to a new action right above "Notes" on L15580? This will be useful to have a place for documentation for how this works in the code.